### ISSUE TITLE
TypeError: ReadableStreamBYOBReader needs a ReadableByteStreamController

### ISSUE THREAD

**Issue by brettwillis** [oven-sh/bun#12908]
### What version of Bun is running?

1.1.21+70ca2b76c

### What platform is your computer?

Darwin 23.5.0 arm64 arm

### What steps can reproduce the bug?

Send request with body to the Bun server, `request.body.getReader()` throws when trying to obtain a `ReadableStreamBYOBReader`.

```ts
// (simplified)
Bun.serve({
  async fetch(req) {
    const reader = await req.body.getReader({ mode: 'byob' }); // <-- TypeError: ReadableStreamBYOBReader needs a ReadableByteStreamController
    // ...
  }
})
```

### What is the expected behavior?

ReadableStream should support byob mode.

### What do you see instead?

TypeError: ReadableStreamBYOBReader needs a ReadableByteStreamController

**Comment by brettwillis** [comment]
Docs claim that `ReadableStream` is fully implemented https://bun.sh/docs/runtime/nodejs-apis#readablestream. Am I the only one encountering this?

**Comment by paperdave** [comment]
i just hit this in the safari browser while working on #15181, which is funny that the top google search for that message is an issue in bun. the bug seems to be something in BYOB streams, since the code sample i had (using streaming `fetch` in the browser) worked in chrome/v8. adding the `jsc` tag here.

**Comment by RealmyTheMan** [comment]
> Docs claim that `ReadableStream` is fully implemented https://bun.sh/docs/runtime/nodejs-apis#readablestream. Am I the only one encountering this?

I'm not using them directly, but rather through the `file-type` package. Regardless, I'm also using Bun, and I believe I have the same problem.

**Comment by Enduriel** [comment]
I just ran into this issue, I'm surprised this isn't higher priority since to my understanding this goes against Bun's claim of ReadableStreamBYOBReader being 'fully implemented' [here](https://bun.com/docs/runtime/nodejs-apis)

Here's a minimal snippet which fails:
```ts
await Bun.write("test.txt", "test file data")
const file = await Bun.file("test.txt")
file.stream().getReader({
	mode: "byob"
})
```

**Comment by real-marshal** [comment]
yeah the docs specifically say that both `ReadableStreamBYOBReader` and `ReadableByteStreamController` are fully implemented, which is clearly false. i had to wrap the stream in node's stream with `Readable.fromWeb()` for it to work with file-type's `fileTypeStream()`, which doesn't feel nice. kind of funny that it's the node specific variant that works, not the web api one.
